src/darwin/Framework/CHIP/MTRLogging.mm

```diff
@@ -0,0 +1,70 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import "MTRLogging_Internal.h"
+
+#import "MTRFramework.h"
+
+#import <algorithm>
+#import <atomic>
+#import <os/lock.h>
+
+using namespace chip::Logging;
+
+static_assert(MTRLogTypeError == (NSInteger) kLogCategory_Error, "MTRLogType* != kLogCategory_*");
+static_assert(MTRLogTypeProgress == (NSInteger) kLogCategory_Progress, "MTRLogType* != kLogCategory_*");
+static_assert(MTRLogTypeDetail == (NSInteger) kLogCategory_Detail, "MTRLogType* != kLogCategory_*");
+
+static os_unfair_lock logCallbackLock = OS_UNFAIR_LOCK_INIT;
+static MTRLogCallback logCallback = nil;
+
+static void MTRLogCallbackTrampoline(const char * component, uint8_t category, const char * format, va_list args)
```

Comment:
"component" is called "module" elsewhere in logging bits.  Should we do that here, or stick with "component"?

Reply:
Hm yeah I was originally trying to align the naming more with `os_log`, but I actually went back and made the category names match the CHIP names. I might call it `moduleName` since `module` is apparently a semi-reserved word in C++ 20 and Xcode highlights it like a keyword.

Reply:
Sounds good, thanks!